### PR TITLE
[Feature] Convert query statement to sql which use ids

### LINKED ISSUE #6914: Convert query statement to sql which use id
## Enhancement
We need save `materialized views` or `views`'s query statement into catalog. Currently we convert query statement to sql  which use names, if db renamed or table renamed, `materialized views` or `views` may have issues. This PR can use ids to replace names.

### PR DESCRIPTION
**Pull request by blackstar-baba**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #6914 

## Problem Summary(Required) ：

Add Utils help query statement convert to sql which use ids.

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/common/util/SqlWithIdUtils.java [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.stream.Collectors.toList;
+
+/**
+ * Help use ids to replace names in sql, it's useful for save statement into catalog
+ */
+public class SqlWithIdUtils {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/common/util/SqlWithIdUtils.java:38]: Could put this class to com.starrocks.sql.common package

Comment (blackstar-baba) [reply fe/fe-core/src/main/java/com/starrocks/common/util/SqlWithIdUtils.java:38]: Ok.

**Comment by blackstar-baba** [comment]
run starrocks_fe_unittest

**Review by kangkaisen — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Comment by kangkaisen** [comment]
run  starrocks_fe_unittest

**Comment by blackstar-baba** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 77 / 113 (68.14%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/common/SqlWithIdUtils.java|77|113|68.14%|

**Pull request merged**